lib/CCreatureHandler.cpp

```diff
@@ -181,6 +181,18 @@ void CCreature::fillWarMachine()
 	warMachine = ArtifactID::NONE; //this creature is not artifact
 }
 
+bool CCreature::DisableChildLinkage = false;
+
+void CCreature::newChildAttached(CBonusSystemNode *child)
+{
+	if(!DisableChildLinkage) CBonusSystemNode::newChildAttached(child);

```

Comment:
This is hack. I am not sure how to do it in a bit more accurate way. The idea is that we need to somehow disable linking temporary bonus nodes from their non-temporary parent nodes. For instance TemporaryArmy has stacks which are also temporary but they are referenced from CCreature which belongs to global three. This backward link looks useless for temporary bonus nodes which leave just a short period of time especially taking into account that CCreature can not (at least should not) change during that period.

Reply:
Probably I can try to introduce some virtual bool isTemporaryNode() to IBonusBearer instead.

Reply:
It does not look convenient to use something like StackWithBonuses. It acts as a proxy while I want to create brand new CArmedInstance to leverage from existing code for morale calculation. And this instance might consist of stacks which does not exist at all like creatures available to buy.

Reply:
Reworked it to temporary bonus node concept